lab_4_retrieval_w_clustering/main.py

```diff
@@ -388,6 +593,11 @@ def make_report(self, num_examples: int, output_path: str) -> None:
             num_examples (int): number of examples for each cluster
             output_path (str): path to output file
         """
+        if not isinstance(output_path, str) or not output_path:
+            raise ValueError('Invalid input argument(s)')
+        clusters_info = self.__algo.get_clusters_info(num_examples)
+        with open(output_path, 'w', encoding='utf-8') as file:
+            dump(clusters_info, file, indent=4)
```

Comment:
what is the difference between ```dump``` and ```dumps``` ?

Reply:
Dump 
- I can save some python object into a json file
- I can't convert a python object into a json equivalent 
Dumps
- I can't save some python object into a json file 
- I can convert a python object into a json equivalent 

+ Different syntax, etc